backend/resell-dapp-api/src/events/event.service.ts

```diff
@@ -3,13 +3,76 @@ import { Injectable } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { Event } from './schemas/event.schema';
 import { CreateEventDto } from './dto/create-event.dto';
+import { TicketDto } from './dto/ticket.dto';
 
 @Injectable()
 export class EventService {
+
   constructor(@InjectModel(Event.name) private eventModel: Model<Event>) {}
 
-  async create(createEventDto: CreateEventDto): Promise<Event> {
+  async createEvent(createEventDto: CreateEventDto): Promise<Event> {
     const createdEvent = new this.eventModel(createEventDto);
     return createdEvent.save();
   }
+
+  async createTicket(ticketDto: TicketDto, contractAddress: string): Promise<Event> {
+    const event = await this.eventModel.findOne({ contractAddress: contractAddress });
+    if (event) {
+      // check if ticket already exists
+      const ticket = event.ticketList.find(ticket => ticket.ticket_address === ticketDto.ticket_address);
+      if (ticket) {
+        throw new Error('Ticket already exists');
+      }
+      // check if the length of the ticket list is equal to the quantity of the event
+      if (event.ticketList.length === event.quantity) {
+        throw new Error('No more tickets available');
+      }
+      
+      event.ticketList.push(ticketDto);
+      return event.save();
+    }
+    else {
+      throw new Error('Event not found');
+    }
+  }
+
+  async resellTicket(ownerAddress: string, quantity: number, contractAddress: string) {
```

Comment:
Yesterday we established that we will have a single endpoint for listing and selling tickets, just a simple update ticket PUT so that we can have clean CRUD without code repetition. Frontend can just change the body it sends.
Here is the documentation on notion.
![image](https://github.com/bangtam1994/Re-sell-dapp/assets/57012729/53cb831d-4815-42f7-acfd-4e6b437bd85b)

We can introduce a new dto `UpdateTicketDto` with these fields